File: api/Models/Comment.cs
using Supabase.Postgrest.Attributes;
using Supabase.Postgrest.Models;

namespace api.Models
{
    [Table("comments")]
    public class Comment : BaseModel
    {
        [PrimaryKey("comment_id", false)]
        public string CommentId { get; set; } = string.Empty;

        [Column("reference_id")]
        public string ReferenceId { get; set; } = string.Empty;

        [Column("reference_type")]
        public string ReferenceType { get; set; } = string.Empty;

        [Column("user_id")]
        public string? UserId { get; set; }

        [Column("parent_comment_id")]
        public string? ParentCommentId { get; set; }

        [Column("content")]
        public string? Content { get; set; }

        [Column("is_edited")]
        public bool IsEdited { get; set; }

        [Column("is_deleted")]
        public bool IsDeleted { get; set; }

        [Column("created_at")]
        public DateTime? CreatedAt { get; set; }

        [Column("updated_at")]
        public DateTime? UpdatedAt { get; set; }
    }

    [Table("comment_mentions")]
    public class CommentMention : BaseModel
    {
        [PrimaryKey("mention_id", false)]
        public string MentionId { get; set; } = string.Empty;

        [Column("comment_id")]
        public string CommentId { get; set; } = string.Empty;

        [Column("mentioned_user_id")]
        public string MentionedUserId { get; set; } = string.Empty;

        [Column("created_at")]
        public DateTime? CreatedAt { get; set; }
    }
}

File: web/src/components/services/comment.ts
import { API_BASE } from './cfg';
import { supabase } from './cfg';

export interface CommentData {
  commentId: string;
  userId: string | null;
  nickname: string | null;
  avatarUrl: string | null;
  parentCommentId: string | null;
  content: string | null;
  isEdited: boolean;
  isDeleted: boolean;
  createdAt: string;
  updatedAt?: string;
}

export const commentApi = {
  // Получить комментарии к указанному объекту
  getComments: async (referenceType: string, referenceId: string): Promise<CommentData[]> => {
    const { data: { session } } = await supabase.auth.getSession();

    const response = await fetch(`${API_BASE}/api/comment/${referenceType}/${referenceId}`, {
      headers: {
        'Authorization': `Bearer ${session?.access_token}`,
        'Content-Type': 'application/json'
      },
    });

    if (!response.ok) throw new Error('Ошибка загрузки комментариев');
    return await response.json();
  },

  // Создать новый комментарий (или ответ на комментарий)
  create: async (dto: {
    referenceType: string;
    referenceId: string;
    content: string;
    parentCommentId?: string;
  }): Promise<CommentData> => {
    const { data: { session } } = await supabase.auth.getSession();

    const response = await fetch(`${API_BASE}/api/comment`, {
      method: 'POST',
      headers: {
        'Authorization': `Bearer ${session?.access_token}`,
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(dto),
    });

    if (!response.ok) {
      const error = await response.json().catch(() => ({}));
      throw new Error(error.message || 'Ошибка создания комментария');
    }
    return await response.json();
  },

  // Обновить (отредактировать) комментарий
  update: async (commentId: string, content: string): Promise<void> => {
    const { data: { session } } = await supabase.auth.getSession();

    const response = await fetch(`${API_BASE}/api/comment/${commentId}`, {
      method: 'PUT',
      headers: {
        'Authorization': `Bearer ${session?.access_token}`,
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({ content }),
    });

    if (!response.ok) {
      const error = await response.json().catch(() => ({}));
      throw new Error(error.message || 'Ошибка редактирования комментария');
    }
  },

  // Удалить комментарий
  delete: async (commentId: string): Promise<void> => {
    const { data: { session } } = await supabase.auth.getSession();

    const response = await fetch(`${API_BASE}/api/comment/${commentId}`, {
      method: 'DELETE',
      headers: {
        'Authorization': `Bearer ${session?.access_token}`,
        'Content-Type': 'application/json'
      },
    });

    if (!response.ok) {
      const error = await response.json().catch(() => ({}));
      throw new Error(error.message || 'Ошибка удаления комментария');
    }
  },
};